Gems/GradientSignal/Code/Source/Components/ImageGradientComponent.cpp

```diff
@@ -550,6 +582,46 @@ namespace GradientSignal
         SetupMultiplierAndOffset(m_configuration.m_scaleRangeMin, m_configuration.m_scaleRangeMax);
     }
 
+    void ImageGradientComponent::HandleSamplingType(float& value, AZ::u32 x, AZ::u32 y, float pixelX, float pixelY) const
+    {
+        switch (m_currentSamplingType)
+        {
+        case SamplingType::Bilinear:
+            // Bilinear interpolation
+            //
+            //   |
+            //   |
+            //   |  (x0,y1) *             * (x1,y1)
+            //   |
+            //   |                o (x,y)
+            //   |
+            //   |  (x0,y0) *             * (x1,y0)
+            //   |___________________________________
+            //
+            // The bilinear filtering samples from a grid around a desired pixel (x,y)
+            // x0,y0 contains one corner of our grid square, x1,y1 contains the opposite corner, and deltaX/Y is the fractional
+            // amount the position exists between those corners.
+            // Ex: (3.3, 4.4) would have a x0,y0 of (3, 4), a x1,y1 of (4, 5), and a deltaX/Y of (0.3, 0.4).
+            const auto& width = m_imageDescriptor.m_size.m_width;
+            const auto& height = m_imageDescriptor.m_size.m_height;
+            float deltaX = pixelX - floor(pixelX);
+            float deltaY = pixelY - floor(pixelY);
+
+            // Don't let the x1,y1 point extend beyond the bounds of our width/height
+            AZ::u32 x1 = AZStd::min(x + 1, width - 1);
+            AZ::u32 y1 = AZStd::min(y + 1, height - 1);
+
```

Comment:
Hmmm, there's an edge case that we might want to look at for a follow-up PR. Interpolation along the edges of the image can be different based on the GradientTransform setting - the image might repeat, mirror, clamp to zero, or clamp to edge. Right now, the logic here is probably causing seams on the boundaries because it isn't taking the wrapping type into account.

Reply:
Good call, will discuss offline the best way to handle this, I think I have an idea